GCC-7 support

Compiling with GCC-7 yields the following error:
```
csvstream.h:113:13: error: this statement may fall through [-Werror=implicit-fallthrough=]
       state = UNQUOTED;
       ~~~~~~^~~~~~~~~~
```

Be sure to update `.travis.yml` to include gcc-7 in the build matrix.

--- a/.travis.yml
+++ b/.travis.yml
@@ -54,15 +54,15 @@ matrix:
           packages: g++-6
       env: MATRIX_EVAL="CXX=g++-6"
 
-    # - name: "Ubuntu 16.04 Xenial GCC 7"
-    #   os: linux
-    #   dist: xenial
-    #   compiler: gcc
-    #   addons:
-    #     apt:
-    #       sources: ubuntu-toolchain-r-test
-    #       packages: g++-7
-    #   env: MATRIX_EVAL="CXX=g++-7"
+    - name: "Ubuntu 16.04 Xenial GCC 7"
+      os: linux
+      dist: xenial
+      compiler: gcc
+      addons:
+        apt:
+          sources: ubuntu-toolchain-r-test
+          packages: g++-7
+      env: MATRIX_EVAL="CXX=g++-7"
 
     - name: "macOS 10.11 Apple LLVM version 8.0.0"
       os: osx

--- a/csvstream.h
+++ b/csvstream.h
@@ -109,8 +109,14 @@ static bool read_csv_line(std::istream &is,
     switch (state) {
     case BEGIN:
       // We need this state transition to properly handle cases where nothing
-      // is extracted.  Note the intended switch fallthrough.
+      // is extracted.
       state = UNQUOTED;
+
+      // Intended switch fallthrough.  Beginning with GCC7, this triggers an
+      // error by default.  Disable the error for this specific line.
+      #if __GNUG__ && __GNUC__ >= 7
+      [[fallthrough]];
+      #endif
 
     case UNQUOTED:
       if (c == '"') {

